### PR TITLE
Fail ios workflow when simctl does not work

### LINKED ISSUE #13063: Confusing Flutter error message when Xcode installation is incomplete
## Steps to Reproduce

Reported on Gitter: During installation of Flutter on a new MacBook, install Xcode, but skip the step of launching the iOS Simulator. Then `simctl` doesn't work, and `flutter doctor` reports e.g.

```text
[✓] iOS toolchain - develop for iOS devices (Xcode 9.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 9.1, Build version 9B55
    • ios-deploy 1.9.2
    • CocoaPods version 1.2.1
```
but then
```text
Error executing simctl: -6
dyld: Library not loaded: /Library/Developer/PrivateFrameworks/CoreSimulator.framework/Versions/A/CoreSimulator
  Referenced from: /Applications/Xcode.app/Contents/Developer/usr/bin/simctl
  Reason: image not found
```
From this error message, it is not very clear that the solution is to launch the Simulator to force the installation of the missing components.

The error message also seems to appear, if iOS tooling isn't set up at all, see https://github.com/flutter/flutter/issues/13030

### PR DESCRIPTION
**Pull request by jonahwilliams**
Fixes #13063 by
  * Partial failure of IOS workflow if `xcrun simctl list` does not exit successfully.  This prints a message that the user must open xcode and install required components.
  * Don't display connected devices if `simctl` is not enabled

### CODE REVIEW

**Review by yjbanov — APPROVED** [review]

**Comment by yjbanov** [comment]
lgtm

File: packages/flutter_tools/lib/src/ios/ios_workflow.dart [hunk 1]
```diff
@@ -87,6 +87,13 @@ class IOSWorkflow extends DoctorValidator implements Workflow {
           'Xcode end user license agreement not signed; open Xcode or run the command \'sudo xcodebuild -license\'.'
         ));
       }
+      if (!xcode.isSimctlInstalled) {
+        xcodeStatus = ValidationType.partial;
+        messages.add(new ValidationMessage.error(
+          'Xcode requires additional components to be installed in order to run.  Launch Xcode and install '
```
Comment (cbracken) [inline packages/flutter_tools/lib/src/ios/ios_workflow.dart]: nit: I'd add a linebreak after the period.

File: packages/flutter_tools/lib/src/ios/mac.dart [hunk 2]
```diff
@@ -149,6 +149,27 @@ class Xcode {
...
+    if (_isSimctlInstalled == null) {
+      try {
+        // This command will error if additional components need to be installed in
+        // xcode 9.2 and above.
+        final ProcessResult result = processManager.runSync(<String>['/usr/bin/xcrun', 'simctl', 'list']);
+        if (result.stderr != null) {
+          _isSimctlInstalled = false;
+        } else {
+          _isSimctlInstalled = true;
+        }
```
Comment (cbracken) [inline packages/flutter_tools/lib/src/ios/mac.dart]: shorter: `_isSimctlInstalled = result.stderr == null;`

**Review by cbracken — APPROVED** [review]

**Comment by cbracken** [comment]
lgtm modulo comments.
lgtm

**Comment by tvolkert** [comment]
This has been identified as the culprit of https://github.com/flutter/flutter/issues/15697 - @jonahwilliams sounds like you're rolling this back?

**Comment by jonahwilliams** [comment]
Rollback failed, I have a forward fix ready to go

**Pull request merged**